v3.0_roadmap.adoc

```diff
@@ -0,0 +1,243 @@
+= Road map for James 3.0
+
+This document aims at dividing existing and upcoming code into the following categories :
+
+ - supported
+ - experimental
+ - unsupported
+
+== Mailbox section
+
+.Feature matrix : mailbox backend
+|===
+|Feature name |Status | Comment
+
+|JPA
+|supported
+|Already existing code
+
+|Memory
+|Unsupported
+|Reference for testing
+
+|Cassandra
+|experimental
+|Essentially we do not know about how performance will behave. Also consistency issues might arise.
+
+|ElasticSearch
+|supported
+|
+
+|HBase
+|unsupported
+|
+
+|JCR
+|unsupported
+|
+
+|Lucene
+|unsupported
+|
+
+|Caching
+|unsupported
+|
+
+|Maildir
+|unsupported
+|Known issue on MailboxPath handling
+
+|===
+
+What being "supported implies for a backend ?
+
+ - Being MPT tested
+ - Deployement tests part of unit tests (ex : cassandra-guice)
+ - Having full support in server/data/data-@backend@
+ - Having its mappers passing generic unit tests for mappers
+ - It should at least support USER_FLAGS and NAMESPACE features on MPT tests
+ - Be able to handle a constant gatling workload for at least 4 hours with constant latency
+ - Be tested with thunderbird (we should write scenari for that)
+
+
+.Feature matrix : mailbox capabilities
+|===
+|Feature name |Status | Comment
+
+|ACL
+|unsupported
+|
+
+|Shared mailbox
+|unsupported
+|
+
+|Quota
+|experimental
+|
+
+|User flags
+|supported
+|
+
+|Distributed events
+|Unsupported
+|
+
+|===
+
+=== Short note about what needs to be DONE
+
+**General** :
+
+ - Define Thunderbird scenari
+ - Write Gatling tests for IMAP
+
+**Cassandra** :
+
+ - We need a Cassandra implementation for MailRepository
```

Comment:
I wonder if we need this at all. There is stuff to store emails in MailRepositories, but nothing to exploit them after this (or not serious... I have seen some frightening mailets...)

You can exploit them for : 
 - see there content
 - redelivering on error
 - collecting spam samples (training models, validating results ?)

As the functionnality is imcomplete, I really wonder if we need any support for MailRepository for now (as more work is needed to make it interesting).